### ISSUE TITLE
Can't create or update a Big Query View with a schema

### ISSUE THREAD

**Issue by avercoutere** [hashicorp/terraform-provider-google#9096]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v0.15.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v3.64.0

### Affected Resource(s)

google_bigquery_table

### Terraform Configuration Files

```tf
provider "google" {
  project = "{{your_gcp_project}}
  region  = "europe-west1"
  zone    = "europe-west1-b"
}

resource "google_bigquery_dataset" "TestDS" {

  dataset_id                      = "Test"
  default_partition_expiration_ms = "0"
  delete_contents_on_destroy      = "false"
  location                        = "EU"

  access {
    role          = "OWNER"
    special_group = "projectOwners"
  }

  access {
    role          = "READER"
    special_group = "projectReaders"
  }

  access {
    role          = "WRITER"
    special_group = "projectWriters"
  }
}

resource "google_bigquery_table" "testViewSchema" {
  dataset_id = google_bigquery_dataset.TestDS.dataset_id
  expiration_time = "0"
  schema = jsonencode(
  [

    {
      "description":"Column 1 Decription",
      "mode":"NULLABLE",
      "name":"Column1",
      "type":"STRING"
    },
    {
      "description":"Column 2 Decription",
      "mode":"NULLABLE",
      "name":"Column2",
      "type":"STRING"
    }
  ]
  )
  table_id = "testViewSchema"

  view {
    query = <<SQL
select "value1" as col1, "value2" as col2
SQL
    use_legacy_sql = "false"
  }
}
```

### Debug Output

https://gist.github.com/avercoutere/8facad5e6b03b0dc42d6eeaf995037d1

### Expected Behavior

Create the Big Query view (table with a view) with the schema.

A solution would be to create/update view in 2 step, first step creating/updating view without schema, second step with applying schema attribute to workaround google api limitation. 

### Actual Behavior

Failed with : "Error: googleapi: Error 400: Schema field shouldn't be used as input with a view, invalid"

The workaround is to first create/update view without schema and launch again terraform with the schema attribute.

### Steps to Reproduce

1. Execute `terraform apply` with the provided tf script with the google provider configured

**Comment by avercoutere** [comment]
I know that issue must be discuss before implementing, but take this PR has a discussion start. Currently the code in this Pull Request is working for my use cases and don't seems to broke anything but i'm sure that you have advice for me.

**Comment by ScottSuarez** [comment]
https://issuetracker.google.com/171967405

Normally we don't do work arounds for api limitations, its a bit of a maintenance burden. We would prefer you escalate with the api to add this functionality. I'll discus with the team to gather thoughts. @rileykarson

**Comment by avercoutere** [comment]
> https://issuetracker.google.com/171967405
> 
> Normally we don't do work arounds for api limitations, its a bit of a maintenance burden. We would prefer you escalate with the api to add this functionality. I'll discus with the team to gather thoughts. @rileykarson

Issue you mentioned said that they won't fix :( The thing is, with Terraform Cloud triggered by VCS, it's impossible to make 2 steps deployment.

**Comment by ScottSuarez** [comment]
I reopened the issue. We'll see if anyone's able to volley it. If I don't get an update I'll try and ping someone later in the week.

**Comment by ScottSuarez** [comment]
Also kind of hacky but you could maybe make a script execution/hook after provisioning to get the resource into your desired state. Not ideal but maybe could unblock you if this is a P0 for you while we figure out the best solution for dealing with this on our end.

**Comment by ScottSuarez** [comment]
It looks like it was assigned to someone, I'll continue to track it on my side..

**Comment by ScottSuarez** [comment]
response from team 
```
We do not have any plans to copy the descriptions from the base tables. Views can contained derived or aggregated columns so there would be a lot of logic to decide whether to copy the description. That being said, we are considering allowing users to specify the description in the CREATE VIEW and ALTER VIEW statements.
```

not much we can do on our side here. sorry folks

**Comment by ScottSuarez** [comment]
actually going to talk about this in our triage to see
